resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -161,6 +170,27 @@ private[spark] abstract class YarnSchedulerBackend(
     totalRegisteredExecutors.get() >= totalExpectedExecutors * minRegisteredRatio
   }
 
+  override def getMergerLocations(
+      numPartitions: Int,
+      resourceProfileId: Int): Seq[BlockManagerId] = {
+    // Currently this is naive way of calculating numMergersNeeded for a stage. In future,
+    // we can use better heuristics to calculate numMergersNeeded for a stage.
+    val tasksPerExecutor = sc.resourceProfileManager
+        .resourceProfileFromId(resourceProfileId).maxTasksPerExecutor(sc.conf)
+    val numMergersNeeded = math.min(
+      math.max(1, math.ceil(numPartitions / tasksPerExecutor).toInt), maxNumExecutors)
```

Comment:
This should also consider case when dynamic allocation is disabled.
`maxNumExecutors` is being set to `DYN_ALLOCATION_MAX_EXECUTORS` - which will become `Int.MaxValue` if DRA is not enabled.

Reply:
Yes, you're right. I was wondering without DRA or without any max executors set for DRA then bounding the max to `Int.MaxValue` is reasonable right? `BlockManagerMasterEndpoint` would only return as many which are available. Thoughts?

Reply:
Without DRA, shouldn't `maxNumExecutors` not be max executors statically configured for the spark application ?
Else we are in effect we are setting `numMergersNeeded` to `numPartitions / tasksPerExecutor` always

Reply:
Yes you're right. I totally forgot about the static executors after having DRA for so long ;)